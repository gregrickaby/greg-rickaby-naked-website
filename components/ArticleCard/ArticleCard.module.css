.card {
  @apply mt-8 overflow-hidden rounded-xl bg-gray-50 shadow-md dark:bg-gray-900;
}

.wrap {
  @apply lg:flex;
}

.imageWrap {
  @apply lg:shrink-0;
}

.image {
  @apply h-48 w-full object-cover lg:h-full lg:w-48;
}

.inner {
  @apply p-8;
}

.meta {
  @apply mb-4 text-sm font-semibold uppercase tracking-wide text-gray-500;
}

.title {
  @apply mt-1 block text-xl font-medium leading-tight hover:underline;
}

.excerpt {
  @apply mt-2 text-base;
}
