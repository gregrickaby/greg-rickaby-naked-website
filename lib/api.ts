import {Post} from '@/lib/types'

/**
 * Get all posts from the WordPress REST API with pagination.
 */
export async function getAllPosts(): Promise<Post[]> {
  try {
    const response = await fetch(
      'https://blog.gregrickaby.com/wp-json/wp/v2/posts?per_page=100&_fields=id,slug,title',
      {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json'
        },
        cache: 'force-cache',
        next: {revalidate: 3600}
      }
    )

    if (!response.ok) {
      throw new Error('Network response was not ok')
    }

    const data: Post[] = await response.json()

    if (data.length === 0) {
      throw new Error('No posts found')
    }

    return data
  } catch (error) {
    console.error(error)
    throw error
  }
}

/**
 * Get a post from the WordPress REST API by slug.
 */
export async function getPostBySlug(slug: string): Promise<Post> {
  // Remove "/blog/" from the slug.
  slug = slug.replace(/^\/blog\//, '')

  try {
    const response = await fetch(
      `https://blog.gregrickaby.com/wp-json/wp/v2/posts?slug=${slug}&_fields=id,date,slug,status,title,excerpt,content,acf,category_names,tag_names,featured_image_data,yoast_head_json`,
      {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json'
        },
        cache: 'force-cache',
        next: {revalidate: 3600}
      }
    )

    if (!response.ok) {
      throw new Error('Network response was not ok')
    }

    const data: Post[] = await response.json()

    if (data.length === 0) {
      throw new Error('Post not found')
    }

    return data[0]
  } catch (error) {
    console.error(error)
    throw error
  }
}
