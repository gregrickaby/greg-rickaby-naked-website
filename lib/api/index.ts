export * from './comments'
export * from './github'
export * from './queries'
export * from './threads'
export * from './WP_Query'
