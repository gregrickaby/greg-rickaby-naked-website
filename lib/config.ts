import {
  IconBook,
  IconBrandAmazon,
  IconBrandBluesky,
  IconBrandCodepen,
  IconBrandFacebook,
  IconBrandFeedly,
  IconBrandGithub,
  IconBrandInstagram,
  IconBrandLinkedin,
  IconBrandThreads,
  IconBrandWordpress,
  IconBrandYoutube,
  IconRss
} from '@tabler/icons-react'

const config = {
  siteName: 'Greg Rickaby',
  siteDescription: 'Cameras and Code',
  siteUrl: 'https://gregrickaby.com',
  siteLogo: '/logo.webp',
  siteOgImage: '/og.webp',
  jobTitle: 'Technical Lead',
  email: 'greg@gregrickaby.com',
  intro: 'Disney Dad. Full-Stack Developer. Photographer. Author.',
  gravatarApiUrl:
    'https://en.gravatar.com/28af3e39c0a1fe4c31367c7e9a8bcac3.json',
  navigation: [
    {
      name: 'About',
      url: '/about'
    },
    {
      name: 'Blog',
      url: '/blog'
    },
    {
      name: 'Contact',
      url: '/contact'
    },
    {
      name: 'Media',
      url: '/media'
    },
    {
      name: 'Resources',
      url: '/resources'
    },
    {
      name: 'Resume',
      url: '/resume'
    }
  ],
  socials: [
    {
      name: 'Bluesky',
      url: 'https://bsky.app/profile/gregrickaby.bsky.social',
      icon: IconBrandBluesky
    },
    {
      name: 'Instagram',
      url: 'https://www.instagram.com/gregoryrickaby/',
      icon: IconBrandInstagram
    },
    {
      name: 'Threads',
      url: 'https://www.threads.net/@gregoryrickaby',
      icon: IconBrandThreads
    },
    {
      name: 'Facebook',
      url: 'https://www.facebook.com/gregrickaby',
      icon: IconBrandFacebook
    },
    {
      name: 'GitHub',
      url: 'https://github.com/gregrickaby/',
      icon: IconBrandGithub
    },
    {
      name: 'LinkedIn',
      url: 'https://linkedin.com/in/gregrickaby/',
      icon: IconBrandLinkedin
    },
    {
      name: 'WordPress',
      url: 'https://profiles.wordpress.org/gregrickaby/',
      icon: IconBrandWordpress
    },
    {
      name: 'Amazon Author',
      url: 'https://www.amazon.com/author/gregrickaby',
      icon: IconBrandAmazon
    },
    {
      name: 'Goodreads',
      url: 'https://www.goodreads.com/author/show/16999736.Greg_Rickaby',
      icon: IconBook
    },
    {
      name: 'CodePen',
      url: 'https://codepen.io/gregrickaby',
      icon: IconBrandCodepen
    },
    {
      name: 'YouTube',
      url: 'https://www.youtube.com/@GregRickaby',
      icon: IconBrandYoutube
    },
    {
      name: 'Feedly',
      url: 'https://feedly.com/i/subscription/feed%2Fhttps%3A%2F%2Fgregrickaby.com%2Ffeed.xml',
      icon: IconBrandFeedly
    },
    {
      name: 'RSS Feed',
      url: 'https://gregrickaby.com/feed.xml',
      icon: IconRss
    }
  ]
}

export default config
