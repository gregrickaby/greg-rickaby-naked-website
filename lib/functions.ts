/**
 * Format a date.
 *
 * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat
 */
export function formatDate(date: Date): string {
  return new Intl.DateTimeFormat('en-US', {
    dateStyle: 'medium',
    timeStyle: 'short'
  }).format(new Date(date))
}
