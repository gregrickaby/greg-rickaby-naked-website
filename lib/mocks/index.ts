export * from './mockPost'
export * from './mockRepos'
