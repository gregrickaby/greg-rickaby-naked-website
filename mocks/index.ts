export * from './mockComments'
export * from './mockPost'
export * from './mockRepos'
