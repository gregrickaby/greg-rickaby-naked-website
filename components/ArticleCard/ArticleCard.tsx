import {formatDate} from '@/lib/functions/formatDate'
import {sanitizeText} from '@/lib/functions/sanitizeText'
import {Post} from '@/lib/types'
import clsx from 'clsx'
import Image from 'next/image'
import Link from 'next/link'
import styles from './ArticleCard.module.css'

/**
 * Article Card component.
 */
export function ArticleCard({post}: Readonly<{post: Post}>) {
  return (
    <Link
      className={clsx('not-prose', styles.card)}
      href={`/blog/${post.slug}`}
      key={post.id}
    >
      <article data-testid="article-card">
        <div className={styles.wrap}>
          <div className={styles.imageWrap}>
            <Image
              alt={post.title.rendered}
              className={styles.image}
              height={192}
              src={post.featured_image_data.url}
              width={594}
            />
          </div>

          <div className={styles.inner}>
            <div className={styles.meta}>
              <time dateTime={post.date}>{formatDate(post.date)}</time>
            </div>

            <h3 className={styles.title}>
              {sanitizeText(post.title.rendered)}
            </h3>

            <div className={styles.excerpt}>
              {sanitizeText(post.excerpt.rendered)}
            </div>
          </div>
        </div>
      </article>
    </Link>
  )
}
