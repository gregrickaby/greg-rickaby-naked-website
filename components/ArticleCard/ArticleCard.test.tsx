import {mockPost} from '@/lib/mocks'
import '@testing-library/jest-dom'
import {render} from '@testing-library/react'
import {axe} from 'jest-axe'
import {describe, expect} from 'vitest'
import {ArticleCard} from './ArticleCard'

describe('ArticleCard Component', () => {
  it('should render an article card', () => {
    const {getByTestId} = render(<ArticleCard post={mockPost} />)
    const articleCard = getByTestId('article-card')
    expect(articleCard).toBeInTheDocument()
  })

  it('should not have any accessibility issues', async () => {
    const {container} = render(<ArticleCard post={mockPost} />)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })

  it('should match snapshot', () => {
    const {getByTestId} = render(<ArticleCard post={mockPost} />)
    const articleCard = getByTestId('article-card')
    expect(articleCard).toMatchSnapshot()
  })
})
