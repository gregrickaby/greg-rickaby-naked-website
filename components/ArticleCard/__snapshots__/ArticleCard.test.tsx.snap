// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ArticleCard Component > should match snapshot 1`] = `
<article
  class="not-prose _card_c0c515"
  data-testid="article-card"
>
  <div
    class="_wrap_c0c515"
  >
    <div
      class="_imageWrap_c0c515"
    >
      <a
        href="/blog/we-went-chasing-waterfalls"
      >
        <img
          alt="We Went Chasing Waterfalls"
          class="_image_c0c515"
          data-nimg="1"
          decoding="async"
          height="192"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fblog.gregrickaby.com%2Fwp-content%2Fuploads%2F2023%2F07%2FDSC9490_Daves-Falls_Amberg_WI_20230711.jpg&w=1200&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fblog.gregrickaby.com%2Fwp-content%2Fuploads%2F2023%2F07%2FDSC9490_Daves-Falls_Amberg_WI_20230711.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fblog.gregrickaby.com%2Fwp-content%2Fuploads%2F2023%2F07%2FDSC9490_Daves-Falls_Amberg_WI_20230711.jpg&w=1200&q=75 2x"
          style="color: transparent;"
          width="594"
        />
      </a>
    </div>
    <div
      class="_inner_c0c515"
    >
      <div
        class="_meta_c0c515"
      >
        <time
          datetime="2023-07-21T17:01:57"
        >
          Jul 21, 2023
        </time>
      </div>
      <a
        class="_title_c0c515"
        href="/blog/we-went-chasing-waterfalls"
      >
        <h3>
          We Went Chasing Waterfalls
        </h3>
      </a>
      <div
        class="_excerpt_c0c515"
      >
        <p>
          This post showcases some photos of waterfalls at Daveâ€™s Falls County Park in Amberg Wisconsin
        </p>
        

      </div>
    </div>
  </div>
</article>
`;
